src/Hosting/Hosting/src/Internal/HostingApplication.cs

```diff
@@ -15,8 +15,8 @@ namespace Microsoft.AspNetCore.Hosting
     internal class HostingApplication : IHttpApplication<HostingApplication.Context>
     {
         private readonly RequestDelegate _application;
-        private readonly IHttpContextFactory _httpContextFactory;
-        private readonly DefaultHttpContextFactory _defaultHttpContextFactory;
+        private readonly IHttpContextFactory? _httpContextFactory;
```

Comment:
Could we always assign this value in the ctor so it's non-null?

Reply:
Hmmm, but it shouldn't be used.

Your change would save some bangs, but I think how it is now is better.